reana_db/utils.py

```diff
@@ -477,3 +559,23 @@ def store_workflow_disk_quota(
         Session.commit()
 
     return workflow_resource
+
+
+def update_workflows_disk_quota() -> None:
+    """Update the disk quotas of all workflows in a more efficient way."""
+    from reana_db.database import Session
+    from reana_db.models import Workflow
+
+    # logs and reana_specification are not loaded to avoid consuming
+    # huge amounts of memory
+    workflows = Workflow.query.options(
+        defer(Workflow.logs), defer(Workflow.reana_specification)
+    ).all()
+    # We expunge all the workflows, as they will not be modified when updating the quotas.
+    # This makes `Session.commit()` much faster
+    for workflow in workflows:
+        Session.expunge(workflow)
+    timer = Timer("Workflow disk", total=len(workflows))
```

Comment:
Cosmetics: the log messages currently look like:

```
2023-08-25 14:22:22,682 | root | MainThread | INFO | Workflow disk progress: 8/8  elapsed: 0.066s  est.total: 0.066s  per event: 0.008s
2023-08-25 14:22:22,703 | root | MainThread | INFO | User disk progress: 3/3  elapsed: 0.019s  est.total: 0.019s  per event: 0.006s
Users disk quota usage updated successfully.
2023-08-25 14:22:22,721 | root | MainThread | INFO | Workflow CPU progress: 8/8  elapsed: 0.015s  est.total: 0.015s  per event: 0.002s
2023-08-25 14:22:22,738 | root | MainThread | INFO | User CPU progress: 3/3  elapsed: 0.016s  est.total: 0.016s  per event: 0.005s
Users cpu quota usage updated successfully.
```

We may want to improve the INFO message (it's not "CPU progress"!) by using something like: 

- "Workflow disk quota usage update progress: 8/8"

and clarify the final message:

- "Disk quota usage updated successfully for all users and workflows."